director/director.go

```diff
@@ -566,12 +566,12 @@ func registerServeAd(engineCtx context.Context, ctx *gin.Context, sType server_s
 	ctx.Set("serverName", adV2.Name)
 	ctx.Set("serverWebUrl", adV2.WebURL)
 
-	ad_url, err := url.Parse(adV2.DataURL)
+	adUrl, err := url.Parse(adV2.DataURL)
 	if err != nil {
 		log.Warningf("Failed to parse %s URL %v: %v\n", sType, adV2.DataURL, err)
 		ctx.JSON(http.StatusBadRequest, server_structs.SimpleApiResp{
 			Status: server_structs.RespFailed,
-			Msg:    fmt.Sprintf("Invalid %s registration", sType),
+			Msg:    fmt.Sprintf("Invalid %s registration. DataURL %s is not a valid URL", sType, adV2.DataURL),
```

Comment:
I don't think `DataURL` is something we expose to caches and origins, is it? I forget exactly how we bootstrap this URL, but I think we can use the error message to point the user to whatever configuration options they need to change to get this working. I know we have `Origin.Url/Cache.Url`, but we probably also set those to some defaults when they're not directly configured.